### ISSUE TITLE
support multi panels

### ISSUE THREAD

**Issue by linsui** [zhanghai/MaterialFiles#181]
Thanks for this grear file manager!

Multi windows has been supported, could you also support multi panels? Thanks again!

**Comment by DMSalesman** [comment]
I second this suggestion. This app is already promising as-is, but power users would surely appreciate more features in "Total Commander" style, especially considering this app's excellent UX.

**Comment by zhanghai** [comment]
I considered multi-panel during initial design, but chose multi-window over multi-panel in the end. The primary reason was that apps should try to use what the OS provides instead of inventing their own, and by implementing multi-window users can already have different windows side-by-side with split screen, while in Q they can also simply swipe on the navigation bar to quickly go to the adjacent window. On the other hand, implementing multi-panel takes up more UI space (there's no space for tabs because of breadcrumb navigation), might affect the overall design, and requires complication into the already complicated UI code, so I just didn't go that way.

**Comment by linsui** [comment]
Thanks for you reply. If you decide not to support multi-panel, it's fine, thanks any way. If you're considering multi-panel, tabs is not necessary, an indicator is enough, just like Amaze file manager.

**Comment by zhanghai** [comment]
Yes indicator style could work, and maybe also a setting to disable it so people who don't need it can hide the indicators. I'll keep this open, just that this feature isn't scheduled in the near feature.

**Comment by DMSalesman** [comment]
Another case where multi-panel would be an advantage is when the display of a device is too small to allow someone to work comfortably in split-screen mode, as too few files would be displayed.

**Comment by magnus-gross** [comment]
I looked up this repo just to propose this same feature request, but after having read your explanation that using the OS features everywhere possible is better, I agree with this sentiment.
After trying out the native nav bar swipe to switch "panels", I actually prefer this native window implementation after all over the custom pane implementation in Amaze File Manager.

Anyway, just wanted to write that I really appreciate you properly considering UI proposals.

P.S.: The navbar swipe is already possible on Android 9, it's just not that polished.

**Comment by ccboxes** [comment]
> Yes indicator style could work, and maybe also a setting to disable it so people who don't need it can hide the indicators. I'll keep this open, just that this feature isn't scheduled in the near feature.
既然指示器不在近期规划，请问是否可以在近期将新建窗口选项移到右下角的浮动菜单内？目前的窗口模式在新建窗口时需要两步，手指也需要移动到屏幕最上方，对于全面屏十分不友好。

**Comment by trymeouteh** [comment]
I would like a tab system of some kind

**Comment by MuntashirAkon** [comment]
This is the only feature which is preventing me from using this app. All major file managers (Solid Explorer, Amaze, File Manager, etc.) have similar feature. I prefer Solid Expoler's style though: both panels are visible on landscape mode, and only a single panel visible on portrait mode at a time (which are interchangeable via swipe). Currently using the ugly [File Manager](https://gitlab.com/axet/android-file-manager) as it's the only file manager on F-Droid that has dual panels.

**Comment by Nemris** [comment]
@zhanghai On Samsung One UI 3.1, based on Android 11, it doesn't appear to be possible to launch two instances of the same app side by side, effectively rendering multi-window unusable.

**Comment by linsui** [comment]
Can't you create a new window from the app menu? It still works on LineageOS 18.1.

**Comment by Nemris** [comment]
I can, but the Samsung workflow seems to still require me to manually pick an app from a list if I want to use split-screen.

Having two instances of the app in the Recents and switching back and forth is possible, but not as practical.

**Comment by linsui** [comment]
Split screen works in LineageOS. Looks like it's Samsung's feature...

**Comment by dmuiX** [comment]
The function new window kind of works at least for me as this feature. Great thanks a lot

**Comment by ZeusOfTheCrows** [comment]
i'd like to throw another vote in for tabs; given the arguments against in this thread (and 256):

> Users can simply have different windows side-by-side with split screen

split screen is a hassle to trigger in newer versions of android - it's not just long-press recents like it used to be. multi-pane would be (e.g.) `menu`→`split panes`, the current system is `menu`→`new window`→`recents`→`app icon`→`split screen`→`select other window`, which is many more steps

it also can't be used in conjunction with another app, [like so](https://imgur.com/Y4kMvz5.png)

> In Q they can also simply swipe on the navigation bar to quickly go to the adjacent window, much like multi-panel.

a) some of us still use 3-button nav*

b) it's still a hassle when dealing with two folders and a second app, say a gallery app for managing photos. one would have to keep going into the recents app to choose the 3rd app

> Tab isn't very possible because the space for it is used for breadcrumbs. ([src](https://github.com/zhanghai/MaterialFiles/issues/265#:~:text=tab%20isn't%20very%20possible%20because%20the%20space%20for%20it%20is%20used%20for%20breadcrumbs.%20for%20multi-panel%2C%20this%20would%20be%20a%20duplicate%20of%20%23181.))

i have a 21:9 phone - i would welcome the option to waste a little vertical space for added functionality

plus, the could be very small (see image of mixplorer above - tabs plus breadcrumbs **plus** two bottom toolbars), especially if using mixplorer's "swipe right on last tab to open a new tab"

*i'm aware double tab recents works, but it's clunkier than swiping between tabs inside the app

**Comment by zhanghai** [comment]
After all these years, I do have a plan to introduce Chrome-like tab management when I have the time to rewrite this app in Compose. I've added a multi-platform compose project skeleton in the `next` branch, however of course it takes much much more time to actually do the re-write.

(The other reasons for the re-write incldue my dissatisfaction with the maintainability of current UI code, and myself wishing to use this app on the web like [filebrowser/filebrowser](https://github.com/filebrowser/filebrowser).)

**Comment by ZeusOfTheCrows** [comment]
that's great news! thank you

i do recommend looking at how mixplorer handles them, as (in my opinion) it's almost perfect*

[holding an item in the sidebar opens it in a new tab, and swiping left from the rightmost tab opens a new one. dual-pane mode is effectively the same, but with two tabs simultaneously](https://i.imgur.com/lJpw5I7.mp4) (for some reason my phone refuses to show touch in the recording)

[options in tab dropdown](https://imgur.com/PAhpjnB.png)

[relevant options in settings](https://imgur.com/D9ZdQmF.png)

*i just wish there was an gesture for closing, but they do close when the app is closed

**Comment by foundObjects** [comment]
+1 for this request, this is my most missed feature from mix and solid explorer. After using other file managers for years I can't stop myself from trying to swipe to the right to get to my second view when I'm moving files around.

Two notes: I'm fine with multi pane being disabled by default, I'm also totally fine with there being no indicator and the two panes only being accessible by left and right swipes. I really only need the feature to see what's going on in two places (source and target) when moving files around.

**Comment by mingcenwei** [comment]
> After all these years, I do have a plan to introduce Chrome-like tab management when I have the time to rewrite this app in Compose. I've added a Compose Multi-Platform project skeleton in the `next` branch, however of course it takes much much more time to actually do the re-write. (And my day job got much busier recent years.)
> 
> (The other reasons for the re-write incldue my dissatisfaction with the maintainability of current UI code, and myself wishing to use this app on the web like [filebrowser/filebrowser](https://github.com/filebrowser/filebrowser).)
> 
> So, no plan right now, but I'm open to it now and it might evetually come.

[Prism File Explorer](https://github.com/Raival-e/Prism-File-Explorer)'s multi-tab interface looks quite nice. Maybe we can implement something similar:

- https://github.com/Raival-e/Prism-File-Explorer/blob/master/assets/image2.png
- https://github.com/Raival-e/Prism-File-Explorer/blob/master/assets/image3.png
